undertone-htb.js

```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
          *
          */
 
-        /* ---------- Process adResponse and extract the bids into the bids array ------------*/
-
         var adResponseDic = {};
-        if (adResponse != null) {
-            for (var i=0; i<adResponse.length; i++) {
-                adResponseDic[adResponse[i].bidRequestId] = adResponse[i];
+        if (adResponse !== null) {
+            for (var idx = 0; idx < adResponse.length; idx++) {
+                adResponseDic[adResponse[idx].bidRequestId] = adResponse[idx];
             }
         }
 
         /* MRA partners receive only one parcel in the array. */
-        for (var i = 0; i < returnParcels.length; i++) {
-            var returnParcel = returnParcels[i];
+        for (var parcelIdx = 0; parcelIdx < returnParcels.length; parcelIdx++) {
+            var returnParcel = returnParcels[parcelIdx];
 
-            /* header stats information */
-            var headerStatsInfo = {
-                sessionId: sessionId,
-                statsId: __profile.statsId,
-                htSlotId: returnParcel.htSlot.getId(),
-                xSlotNames: [returnParcel.xSlotName]
-            };
+            /* Header stats information */
+            var headerStatsInfo = {};
+            var htSlotId = returnParcel.htSlot.getId();
+            headerStatsInfo[htSlotId] = {};
+            headerStatsInfo[htSlotId][returnParcel.requestId] = [returnParcel.xSlotName];
 
             returnParcel.targetingType = 'slot';
             returnParcel.targeting = {};
-            returnParcel.targeting[__baseClass._configs.targetingKeys.id] = [returnParcel.requestId];
             returnParcel.pass = false;
 
-            var currAdResponse = adResponseDic[returnParcel.requestId];
-            if (currAdResponse == null) {
+            var currAdResponse = adResponseDic[returnParcel.xSlotRef.bidId];
+            if (currAdResponse === null) {
                 currAdResponse = {};
             }
 
-            if (currAdResponse.ad == null || currAdResponse.cpm <= 0) {
+            if (currAdResponse.ad === null || currAdResponse.cpm <= 0) {
```

Comment:
This `currAdResponse.ad === null` should be updated to `currAdResponse.ad === undefined` as the property `ad` will be undefined in the case where there was no match for `currAdResponse`

Reply:
I can't do that because when there's your lint check does not allow to write undefined

Reply:
Please use `typeof currAdResponse.ad === 'undefined'` instead then.